src/main/scala/Main.scala

```diff
@@ -1,8 +1,84 @@
+
+
+import newmodel.Package
+import newmodel._
+
+import scala.collection.mutable.ListBuffer
+
 /**
  * Created by yaroslav on 11.05.2015.
  */
 object Main {
-  def main(args:Array[String]) ={
-    println("hello world")
+
+  object DummyIndex extends Index{
+    val docElements = ListBuffer.empty[DocElement]
+    override def getConversionsFor(id: ID): Seq[ImplicitConversions] = ???
+
+    override def getKnownSubclasses(id: ID): Seq[ID] = ???
+
+    override def getShadowedConversionsFor(id: ID): Seq[ImplicitConversions] = ???
+
+    override def getLinearSupertypesBy(id: ID): Seq[ID] = ???
+
+    override def getCompanionFor(id: ID): ID = ???
+
+    override def getDocElementBy(id: ID): DocElement = docElements.find((x) => x.id == id).get
```

Comment:
I'd actually rather see client code do `docElements.find(_.id == id))` rather than `getDocElementBy(id)`. its ok for client code to operate on the list. this like an unnecessary indirection. But it is interesting. As I said, I wasn't completely sure building this intermediate representation is the way to go. It still might be, but this looks very much we are building a stripped down Scala AST with helpers to make it easy to navigate it for docs purposes. wdyt? /cc 
@clhodapp